src/openfermion/ops/_polynomial_tensor.py

```diff
@@ -25,64 +25,58 @@ class PolynomialTensorError(Exception):
     pass
 
 
-def one_body_basis_change(one_body_tensor, rotation_matrix):
-    """Change the basis of an 1-body interaction tensor such as the 1-RDM.
+def general_basis_change(general_tensor, rotation_matrix, key):
+    """Change the basis of an general interaction tensor.
 
-    M' = R^T.M.R where R is the rotation matrix, M is the 1-body tensor
-    and M' is the transformed 1-body tensor.
+    M'^{p_1p_2...p_n} = R^{p_1}_{a_1} R^{p_2}_{a_2} ...
+                        R^{p_n}_{a_n} M^{a_1a_2...a_n}
+
+    where R is the rotation matrix, M is the general tensor, M' is the
+    transformed general tensor, and a_k and p_k are indices. The formula uses
+    the Einstein notation (implicit sum over repeated indices).
+
+    In case R is complex, the k-th R in the above formula need to be conjugated
+    if key has a 1 in the k-th place (meaning that the corresponding operator
+    is a creation operator).
 
     Args:
-        one_body_tensor: A square numpy array or matrix containing information
-            about a 1-body interaction tensor such as the 1-RDM.
+        general_tensor: A square numpy array or matrix containing information
+            about a general interaction tensor.
         rotation_matrix: A square numpy array or matrix having dimensions of
-            n_qubits by n_qubits. Assumed to be real and invertible.
+            n_qubits by n_qubits. Assumed to be unitary.
+        key: A tuple indicating the type of general_tensor. Assumed to be
+            non-empty.
 
     Returns:
-        transformed_one_body_tensor: one_body_tensor in the rotated basis.
+        transformed_general_tensor: general_tensor in the rotated basis.
     """
     # If operator acts on spin degrees of freedom, enlarge rotation matrix.
     n_orbitals = rotation_matrix.shape[0]
-    if one_body_tensor.shape[0] == 2 * n_orbitals:
+    if general_tensor.shape[0] == 2 * n_orbitals:
         rotation_matrix = numpy.kron(rotation_matrix, numpy.eye(2))
 
-    # Effect transformation and return.
-    transformed_one_body_tensor = numpy.einsum('qp, qr, rs',
-                                               rotation_matrix,
-                                               one_body_tensor,
-                                               rotation_matrix)
-    return transformed_one_body_tensor
+    order = len(key)
+    if order > 26:
```

Comment:
Where is this order 26 limit coming from? I suggest not having a maximum order restriction.

Reply:
It comes from using the `numpy.einsum` function, which uses letters for indices, and there are 26 upper case and 26 lower case in total (and I need two letters for each dimension of the tensor). However, actually numpy complains about "too many subscripts in einsum" before the 26 limit is reached.

I don't think anything more than 16 is useful in any case.